web/src/message_list.ts

```diff
@@ -274,34 +294,31 @@ export class MessageList {
         }
     }
 
-    selected_id() {
+    selected_id(): number {
         return this.data.selected_id();
     }
 
-    select_id(id, opts) {
-        opts = {
+    select_id(id: number | string, select_id_opts?: SelectIdOpts): void {
+        if (typeof id === "string") {
+            blueslip.warn("Call to select_id with string id");
```

Comment:
Is this something that we expect to have happen? We might want to do a `blueslip.error`, which logs to sentry, so we can find and purge such bugs...

Reply:
I didn't spend the time to see if we should expect it to happen, since `convert_id` was being called freely before. I can change it to an `error` if we're down to invest in fixing any calls with strings.

Reply:
Yeah, that seems worth cleaning up.